### PR TITLE
Feature/join tasks with cards

### PR DESCRIPTION
**Pull request by gainullin81**

### CODE REVIEW

File: HTTP:
Comment (eugeek) [inline HTTP::1]: remove this one, please 😉

File: app/src/components/Card/CardPreview/index.tsx [hunk 1]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
   const minutes =
     createdAt.getMinutes() < 10 ? '0' + createdAt.getMinutes() : createdAt.getMinutes()
 
+  
+  console.log(card)
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:43]: This one also should be removed after use

File: app/src/components/Card/CardPreview/index.tsx [hunk 2]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
+        <div className="flex-grow overflow-hidden">
+          {!isEmpty(tasks) ? (
+            <div className="flex flex-col gap-3">
+              <div className="col-span-3 overflow-hidden">
+                {tasks.slice(0, 3).map((task) => (
+                  <div key={'prevtask-' + task.id} className="h-[46px] py-1 justify-start items-start gap-2 inline-flex">
+                    <div className="w-4 h-4 justify-center items-center flex">
+                      <div className="w-4 h-4 pl-[1.60px] pr-[1.23px] pt-[1.60px] pb-[1.30px] justify-center items-center inline-flex">
+                        <div className="w-[13.17px] h-[13.10px] relative">
+                          {/* Здесь можно вставить содержимое, например, иконку */}
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:76]: Copilot comments also should be removed 😁

Comment (eugeek) [reply app/src/components/Card/CardPreview/index.tsx:76]: And yes, the icon should be there

File: app/src/components/Card/CardPreview/index.tsx [hunk 3]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
-        {/* <div className='grid grid-cols-5 mt-5'> */}
-        <div className='flex mt-5 sm:max-w-[390px] md:max-w-[300px] lg:max-w-[450px] '>
-          {!isEmpty(imageUrls) ? (
-            <div className='flex col-span-2 '>
-              {imageUrls?.map((url, idx) => (
-                <Image key={'image-' + url + '-' + idx} className='w-12' src={url} alt='' />
+        <div className="mt-auto w-full">
+          <div className="h-[56px] w-full pr-2 bg-white rounded-sm justify-start items-center gap-1 inline-flex">
+            <div className="justify-start items-center flex w-full">
+              {(imageUrls ?? []).slice(0, 4).map((url, index) => (
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:110]: Actually we split this array on 29 line. Could you check my improvements in 29 line 👆

File: app/src/components/Card/CardPreview/index.tsx [hunk 4]
```diff
@@ -28,7 +28,6 @@ const months = [
 function CardPreview({ card, onClick }: ICardPreviewProps) {
   const imageUrls = card?.files
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:29]:
```typescript
const imageUrls = useMemo(() => (card?.files?.filter((file) => file.type === 'image')?.slice(0, 4)?.map((file) => file.url)) ?? [], [card?.files])
```

File: app/src/components/Card/CardPreview/index.tsx (hunk 3 above)
Comment (eugeek) [reply app/src/components/Card/CardPreview/index.tsx:110]: And if we have imageUrls === [] (0 length)  - this means that if there are no images, this block should not be rendered at all.

File: app/src/components/Card/CardPreview/index.tsx [hunk 5]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
+                  </span>
+                </div>
+              )}
             </div>
-          ) : null}
+          </div>
+          <div className="mt-2 flex justify-start items-center gap-2">
+            <AvatarGroup>
+              {(card.users as IUser[])?.map((user, index) => (
+                <Avatar key={index} src={user.avatarUrl} alt={user.name} radius="xl" size="sm" />
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:125]: The key should be an unique value (for example user id)

Comment (eugeek) [reply app/src/components/Card/CardPreview/index.tsx:125]: Actually we have [component ](https://github.com/prspktvs/dun/blob/d1791944525a763d1894a0e69e0d6319eb89e150/app/src/components/ui/Avatar.tsx)for avatars

File: app/src/utils/api.ts [hunk 6]
```diff
@@ -1,20 +1,28 @@
-const API_HOST = process.env.VITE_BACKEND_URL || 'https://api.dun.wtf'
+const API_HOST = process.env.VITE_BACKEND_URL || 'https://api.dun.wtf';
+
+// Получаем токен и UID из localStorage
+const token = localStorage.getItem('token');
+const uid = localStorage.getItem('uid'); // Предполагаем, что UID уже сохранен в localStorage
```
Comment (eugeek) [inline app/src/utils/api.ts:5]: Comments again + we don't use uid 😉

File: app/src/components/Card/CardPreview/index.tsx [hunk 7]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
+            <div className="justify-start items-center flex w-full">
+              {(imageUrls ?? []).slice(0, 4).map((url, index) => (
+                <img key={index} className="w-14 h-14 relative rounded-sm border border-[#efefef]" src={url} />
               ))}
-              {card?.files?.length > 5 ? (
-                <span className='ml-3 flex items-center h-full font underline text-sm ml-12'>
-                  +{card.files.length - 5}
-                </span>
-              ) : null}
+              {(imageUrls ?? []).length > 4 && (
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:113]: The same if imageUrls upper is empty - this block won't be rendered

File: app/src/utils/api.ts [hunk 8]
```diff
@@ -1,20 +1,28 @@
...
-    console.error('Fetch error: ', e)
-    throw e
+    console.error('Fetch error: ', e);
+    throw e;
   }
-}
\ No newline at end of file
+}
+
+// Пример использования UID
```
Comment (eugeek) [inline app/src/utils/api.ts:26]: Also removed

File: backend/src/api/cards.js [hunk 9]
```diff
@@ -186,3 +186,29 @@ export const unshareCard = async (req, res) => {
     res.status(500).send('Internal server error')
   }
 }
+
+// New endpoint for retrieving cards with tasks
+export const getAllCardsWithTasks = async (req, res) => {
+  try {
+    const projectId = req.params.projectId; 
+    const query = `
```
Comment (eugeek) [inline backend/src/api/cards.js:194]: 1. Queries should be in `queries.js`
2. I asked you to expand the existed query by getting tasks.
3. This query has another result - not we expected. I mean that we need to add all tasks with card_id = card.id to this card.
Smth like that:
```sql
SELECT 
    c.*, 
    json_group_array(
        json_object(
            'id', t.id,
            'isDone', t.isDone,
            'text', t.text,
            'users', t.users,
            'author', t.author,
            'priority', t.priority,
            'status', t.status
        )
    ) AS tasks
FROM 
    cards c
LEFT JOIN 
    tasks t ON c.id = t.card_id
GROUP BY 
    c.id;
```

File: app/src/utils/api.ts [hunk 10]
```diff
@@ -1,20 +1,28 @@
-const API_HOST = process.env.VITE_BACKEND_URL || 'https://api.dun.wtf'
+const API_HOST = process.env.VITE_BACKEND_URL || 'https://api.dun.wtf';
+
+// Получаем токен и UID из localStorage
+const token = localStorage.getItem('token');
```
Comment (eugeek) [inline app/src/utils/api.ts:4]: This one is good! But the constants looks better in uppercase 😊

File: app/src/components/Card/CardPreview/index.tsx [hunk 11]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
+                <div className="w-14 h-14 bg-white rounded-sm border border-[#efefef] justify-center items-center flex">
+                  <span className="text-[#46434e] text-sm font-normal font-['Monaspace Argon Var']">
+                    +{(imageUrls ?? []).length - 4}
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:116]:
```typescript
+{(imageUrls ?? []).length - MAX_IMAGES}
```

File: app/src/components/icons/index.tsx [hunk 12]
```diff
@@ -56,50 +56,38 @@ export const UnreadMarker = () => (
...
     <text
```
Comment (eugeek) [inline app/src/components/icons/index.tsx:82]: If we have no text - let's remove the text block 🧐

File: backend/src/api/cards.js [hunk 13]
```diff
@@ -186,3 +186,29 @@ export const unshareCard = async (req, res) => {
     res.status(500).send('Internal server error')
   }
 }
+
+// New endpoint for retrieving cards with tasks
```
Comment (eugeek) [inline backend/src/api/cards.js:190]: No needed new endpoint

File: backend/src/api/tasks.js [hunk 14]
```diff
@@ -1,20 +1,60 @@
-import { allQuery } from '../database/index.js'
-import { SELECT_USER_TASKS_QUERY } from '../database/queries.js'
+import { runQuery, allQuery } from '../database/index.js';
+import { SELECT_USER_TASKS_QUERY } from '../database/queries.js';
+
+export const createTask = async (req, res) => {
```
Comment (eugeek) [inline backend/src/api/tasks.js:4]: I'm not get it for what 😊

File: backend/src/api/tasks.js [hunk 15]
```diff
@@ -1,20 +1,60 @@
...
-    res.json({ tasks })
+    const tasks = await allQuery(SELECT_USER_TASKS_QUERY, [projectId, `%${userId}%`]);
+    res.json({ tasks });
+  } catch (error) {
+    console.error(error);
+    res.status(500).send('Internal server error');
+  }
+};
+
+export const getTasksWithCards = async (req, res) => {
```
Comment (eugeek) [inline backend/src/api/tasks.js:38]: And for what this one? 😯

File: backend/src/server.js [hunk 16]
```diff
@@ -1,118 +1,129 @@
...
-import { CREATE_TABLES_QUERIES, CREATE_ALL_INDEXES } from './database/queries.js'
-import createPushAPI from './push.js'
-import { getSendPushToChatFn } from './api/chats.js'
-
+import express from 'express';
+import cors from 'cors';
+import expressWs from 'express-ws';
+import { onStoreDocument } from './utils/util.js';
+import { Server as HocusPocusServer } from '@hocuspocus/server';
+import routes from './routes.js';  // Подключение маршрутов
```
Comment (eugeek) [inline backend/src/server.js:6]: Remove comment again ╰(*°▽°*)╯

File: backend/src/routes.js [hunk 17]
```diff
@@ -9,21 +10,25 @@ import {
...
+} from './api/cards.js';
 
-const router = express.Router()
+const router = Router();
 
 // tasks
-router.get('/tasks', getUserTasks)
+router.get('/tasks', getUserTasks);
+router.post('/tasks', createTask);
+router.get('/tasks-with-cards', getTasksWithCards);  // Новый маршрут для получения задач с карточками
```
Comment (eugeek) [inline backend/src/routes.js:21]: There are no needed. But if we need to create such endpoint - we could expand the existed one 😉(Not add a new one for each feature or interaction with entity)

File: backend/src/server.js [hunk 18]
```diff
@@ -1,118 +1,129 @@
...
+import { runQuery, sqliteExtension } from './database/index.js';
+import { CREATE_TABLES_QUERIES, CREATE_ALL_INDEXES } from './database/queries.js';
+import createPushAPI from './push.js';
+import { getSendPushToChatFn } from './api/chats.js';
+
+// Парсинг JWT токена
```
Comment (eugeek) [inline backend/src/server.js:12]: also...

File: backend/src/database/ExtendedSQLite.js [hunk 19]
```diff
@@ -1,16 +1,29 @@
...
+    super(configuration);
+    this.initializeDatabase();
   }
 
   async onConfigure() {}
 
   initializeDatabase() {
-    this.db = new sqlite3.Database(this.configuration.database)
-    this.db.run(this.configuration.schema)
+    this.db = new sqlite3.Database(this.configuration.database, (err) => {
```
Comment (eugeek) [inline backend/src/database/ExtendedSQLite.js:14]: Great thing btw, thanks for added.

But it won't work, because you've skipped the second argument (MODE).

File: backend/src/database/ExtendedSQLite.js [hunk 20]
```diff
@@ -1,16 +1,29 @@
...
+      if (err) {
+        console.error('Could not connect to database', err);
+      } else {
+        console.log('Connected to database');
+        this.db.run(this.configuration.schema, (err) => {
```
Comment (eugeek) [inline backend/src/database/ExtendedSQLite.js:19]: `this.db` won't be initialized here

File: backend/src/database/index.js [hunk 21]
```diff
@@ -1,46 +1,55 @@
...
+const schema = `
+  ${CREATE_CARDS_TABLE_QUERY};
+  ${CREATE_TASKS_TABLE_QUERY};
+  ${CREATE_ALL_INDEXES.join('; ')}
+`;
 
 export const sqliteExtension = new ExtendedSQLite({
   database: 'sqlite.db',
-})
+  schema: schema,
```
Comment (eugeek) [inline backend/src/database/index.js:13]: Are you sure that it's needed here? 🧐

File: app/src/components/Card/CardPreview/index.tsx [hunk 22]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
+        </div>
+        <div className="flex-grow overflow-hidden">
+          {!isEmpty(tasks) ? (
+            <div className="flex flex-col gap-3">
+              <div className="col-span-3 overflow-hidden">
+                {tasks.slice(0, 3).map((task) => (
+                  <div key={'prevtask-' + task.id} className="h-[46px] py-1 justify-start items-start gap-2 inline-flex">
+                    <div className="w-4 h-4 justify-center items-center flex">
+                      <div className="w-4 h-4 pl-[1.60px] pr-[1.23px] pt-[1.60px] pb-[1.30px] justify-center items-center inline-flex">
+                        <div className="w-[13.17px] h-[13.10px] relative">
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:75]: what are the exact values? I mean it's okay if it was w-[13px] h-[13px]

File: app/src/components/Card/CardPreview/index.tsx [hunk 23]
```diff
@@ -40,56 +39,97 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
+                          {/* Здесь можно вставить содержимое, например, иконку */}
+                        </div>
+                      </div>
+                    </div>
+                    <div className="grow shrink basis-0">
+                      <span className="text-[#46434e] text-sm font-normal font-['Rubik'] leading-[18.90px]">
+                        {task.title}
+                      </span>
+                      <span className="text-[#46434e] text-sm font-semibold font-['Rubik'] leading-[18.90px]">
+                        @MattD
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:85]: It was a plug (here should be a real value of mention)

**Comment by eugeek** [comment]
I understand that while you are rolling in, there are a lot of new things in the project for you 😉 You've done quite a lot - you're great, but I wrote to you that it would be possible to improve and the best practices here.

But first, **SET UP THE FORMATTER**, please (it's added a lot of unnecessary semicolons at the end of each line)

**Comment by vlad-x** [comment]
@gainullin81 please make sure all comments are in English, thank you!

**Comment by gainullin81** [comment]
Привет, Влад. Да конечно, все исправляю. Юджин указал на это в Ревью.

Пн 28.10.2024 в 14:09, Vlad ***@***.***>:

File: app/src/components/Card/CardPreview/index.tsx [hunk 24]
```diff
@@ -49,87 +36,114 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
-              <div className="self-stretch text-[#46434e] text-xl font-medium font-['Rubik'] leading-7 overflow-hidden" style={{ display: '-webkit-box', WebkitLineClamp: 2, WebkitBoxOrient: 'vertical', overflow: 'hidden', textOverflow: 'ellipsis' }}>
+            <div className='h-[62px] w-full flex-col justify-start items-start gap-3 inline-flex'>
+              <div
+                className="self-stretch text-[#46434e] text-xl font-medium font-['Rubik'] leading-7 overflow-hidden"
+                style={{
+                  display: '-webkit-box',
+                  WebkitLineClamp: 2,
+                  WebkitBoxOrient: 'vertical',
+                  overflow: 'hidden',
+                  textOverflow: 'ellipsis',
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:56]: can we remove those properties?🙂

File: app/src/components/Card/CardPreview/index.tsx [hunk 25]
```diff
@@ -4,43 +4,30 @@ import { Image, Avatar, AvatarGroup } from '@mantine/core'
 import TaskPreview from '../../Task/TaskPreview'
 import { MessageIcon, UpdateIcon } from '../../icons'
 import ProjectUsers from '../../User/ProjectUsers'
+import { useMemo } from 'react'
+import { formatDistanceToNow } from 'date-fns'
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:8]: If you use the package, you need to install it. Btw you have an error in the build because of this 🫠

File: .idea/.gitignore
Comment (eugeek) [inline .idea/.gitignore:1]: No, no, no for any IDE's folders. And you can add this folder to `.gitignore` btw

File: app/src/components/Card/CardPreview/index.tsx [hunk 26]
```diff
@@ -49,87 +36,114 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
+                  >
+                    <div className='w-4 h-4 justify-center items-center flex'>
+                      <div className='w-4 h-4 pl-[1.6px] pr-[1.23px] pt-[1.6px] pb-[1.3px] justify-center items-center inline-flex'>
+                        <i className='ri-checkbox-line text-14' />
                       </div>
                     </div>
-                    <div className="grow shrink basis-0">
-                      <span className="text-[#46434e] text-sm font-normal font-['Rubik'] leading-[18.90px]">
+                    <div className='grow shrink basis-0'>
+                      <span className="text-[#46434e] text-sm font-normal font-['Rubik'] leading-[18.9px]">
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:79]: Why are there single quotes in the first line, but there are double quotes in the second one 🤔I guess smth wrong with prettier again on your side. I will come back and set up eslint finally

File: app/src/components/Card/CardPreview/index.tsx [hunk 27]
```diff
@@ -49,87 +36,114 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
-                <div key={'description-' + line + index} className="text-[#46434e] text-sm font-normal font-['Rubik'] leading-tight overflow-hidden" style={{ display: '-webkit-box', WebkitLineClamp: 2, WebkitBoxOrient: 'vertical', overflow: 'hidden', textOverflow: 'ellipsis' }}>
+                <div
+                  key={'description-' + line + index}
+                  className="text-[#46434e] text-sm font-normal font-['Rubik'] leading-tight overflow-hidden"
+                  style={{
+                    display: '-webkit-box',
+                    WebkitLineClamp: 2,
+                    WebkitBoxOrient: 'vertical',
+                    overflow: 'hidden',
+                    textOverflow: 'ellipsis',
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:106]: the same 👆

File: app/src/components/Card/CardPreview/index.tsx [hunk 28]
```diff
@@ -4,43 +4,30 @@ import { Image, Avatar, AvatarGroup } from '@mantine/core'
...
+    () =>
+      card?.files
+        ?.filter((file) => file.type === 'image')
+        ?.slice(0, MAX_IMAGES)
+        ?.map((file) => file.url) ?? [],
+    [card?.files],
+  )
 
   const countChats = card?.chatIds?.length || 0
   const tasks = card?.tasks
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:28]: We could slice tasks here and use `useMemo`

File: app/src/components/Card/CardPreview/index.tsx [hunk 29]
```diff
@@ -49,87 +36,114 @@ function CardPreview({ card, onClick }: ICardPreviewProps) {
...
-                  <span className="text-[#46434e] text-sm font-normal font-['Monaspace Argon Var']">
-                    +{(imageUrls ?? []).length - 4}
-                  </span>
-                </div>
-              )}
+        {imageUrls.length > 0 && (
+          <div className='mt-auto w-full'>
+            <div className='h-[56px] w-full pr-2 bg-white rounded-sm justify-start items-center gap-1 inline-flex'>
+              <div className='justify-start items-center flex w-full'>
+                {imageUrls.slice(0, MAX_IMAGES).map((url, index) => (
```
Comment (eugeek) [inline app/src/components/Card/CardPreview/index.tsx:119]: Not needed to slice them (because they was sliced in the useMemo)

File: app/src/types/Task.d.ts [hunk 30]
```diff
@@ -17,6 +17,8 @@ export enum TaskPriority {
   Urgent = 'Urgent',
 }
 export interface ITask extends PartialBlock {
+  title: ReactNode
+  assignedTo: ReactNode
```
Comment (eugeek) [inline app/src/types/Task.d.ts:21]: title = text, assignedTo = users 🤔

**Comment by eugeek** [comment]
@gainullin81 Actually, you don't need to create multiple pull requests, we could do with one 🙂

File: backend/src/api/tasks.js (hunk 14 above)
Comment (vlad-x) [reply backend/src/api/tasks.js:4]: @gainullin81 this is no needed, tasks are created in `onStoreDocument` in utils.js – tasks shouldn't be created separately from cards.

File: backend/src/api/tasks.js (hunk 15 above)
Comment (vlad-x) [reply backend/src/api/tasks.js:38]: @gainullin81 I don't think this what needed to be done – as far I remember we need to show tasks inside cards in the grid view, not the other way round? 🤔

File: backend/src/database/index.js [hunk 31]
```diff
@@ -1,46 +1,53 @@
-import { ExtendedSQLite } from './ExtendedSQLite.js'
+import { ExtendedSQLite } from './ExtendedSQLite.js';
+import { CREATE_TASKS_TABLE_QUERY, CREATE_CARDS_TABLE_QUERY, CREATE_ALL_INDEXES } from './queries.js';
+
+const schema = `
+  ${CREATE_CARDS_TABLE_QUERY};
+  ${CREATE_TASKS_TABLE_QUERY};
+  ${CREATE_ALL_INDEXES.join('; ')}
+`;
```
Comment (vlad-x) [inline backend/src/database/index.js:9]: @gainullin81 It does not look like this is needed here?

File: backend/src/routes.js [hunk 32]
```diff
@@ -1,29 +1,41 @@
...
+  const { includeCards } = req.query;
+
+  if (includeCards === 'true') {
+    return getTasksWithCards(req, res);
+  }
+
+  return getUserTasks(req, res);
+});
+
+router.post('/tasks', createTask);
```
Comment (vlad-x) [inline backend/src/routes.js:28]: This is not needed as mentioned above

File: backend/src/server.js [hunk 33]
```diff
@@ -1,118 +1,115 @@
...
 
-const app = express()
-expressWs(app)
 const hocusPocusServer = HocusPocusServer.configure({
   onStoreDocument: async (data) =>
     onStoreDocument({ data, broadcast: { sendMessageToProject, sendMessageToUser } }),
   async onAuthenticate(data) {
-    const { token } = data
-
-    // @TODO: verify token https://firebase.google.com/docs/auth/admin/verify-id-tokens#web
```
Comment (vlad-x) [inline backend/src/server.js:27]: @gainullin81 why was this removed? this comment is meant to be here

Comment (vlad-x) [reply backend/src/server.js:27]: @gainullin81 please restore all removed @TODO comments

File: backend/src/server.js [hunk 34]
```diff
@@ -1,118 +1,115 @@
...
-const sendNotification = createPushAPI(app, '/push/')
-const sentPushToChat = getSendPushToChatFn(sendNotification)
+const sendNotification = createPushAPI(app, '/push/');
+const sendPushToChat = getSendPushToChatFn(sendNotification);
 
-app.post('/internal/chat/:chatId', sentPushToChat)
+app.post('/internal/chat/:chatId', sendPushToChat);
 
 export function sendMessageToUser(userId, message) {
-  // @TODO: add all other types of notifications
```
Comment (vlad-x) [inline backend/src/server.js:74]: @gainullin81 here as well

File: app/src/components/Card/CardPreview/index.tsx (hunk 25 above)
Comment (vlad-x) [reply app/src/components/Card/CardPreview/index.tsx:8]: @gainullin81 make sure date-fns and all new packages are committed in package.json

**Review by vlad-x — CHANGES_REQUESTED** [review]
- Go through the comments and fix the issues, remove redundant code 
- Cleanup files, add .idea etc. to .gitignore and remove it from the PR
- Restore @TODO comments, they are needed

**Comment by vlad-x** [comment]
closing in favor of #41
